File: python/allergies/allergies.py
class Allergies(object):

    ALLERGY_KEYS = ("cats", "pollen", "chocolate", "tomatoes", "strawberries",
                    "shellfish", "peanuts", "eggs")

    def __init__(self, score):
        # Use equivalence to 8-bit strings, where 1 in the ith bit means
        # person has allergy at ALLERGY_KEYS[i].
        allergy_values = [bool(int(x)) for x in format(score, '08b')][-8:]
        self.allergies = dict(zip(self.ALLERGY_KEYS, allergy_values))

    def is_allergic_to(self, item):
        return self.allergies[item]

    @property
    def lst(self):

        return [allergy for (allergy, hasAllergy) in self.allergies.items() if hasAllergy]

File: javascript/allergies/allergies.js
const ALL_ALLERGIES = new Map([
  ['eggs', 1 ], 
  ['peanuts', 2 ], 
  ['shellfish', 4 ], 
  ['strawberries', 8 ], 
  ['tomatoes', 16 ],
  ['chocolate', 32 ],
  ['pollen', 64 ],
  ['cats', 128 ] 
]);

export class Allergies {

  constructor(allergyScore) {

    this.allergies = [...ALL_ALLERGIES].reduce(
      
      (acc, [allergy, value]) => {
        // Bitwise and because each allergy is represented by a
        // different bit.
        if (value & allergyScore)
          acc.push(allergy)

        return acc;
      }, 
      
      []);
  }

  list() { return this.allergies; }

  allergicTo(allergy) { return this.allergies.includes(allergy); }
}
